src/app/signup/page.tsx

```diff
@@ -50,74 +68,141 @@ const SignUp = () => {
           </div>
           <div className="flex items-center px-12 mb-10 w-full">
             <form
-              // @ts-expect-error Type mismatch
-              onSubmit={handleSubmit((data) => signupUser(data))}
+              ref={formRef}
+              onSubmit={handleSubmit(async (data) => {
+                const error = await signupUser(data);
```

Comment:
We should use this pattern instead:

```
  formAction(new FormData(formRef.current!));
```

This ensures the "hidden" server action inputs get included.

Reply:
Alright. I passed in a FormData instead.